sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -130,19 +139,30 @@ def _csr_mean_variance_axis0(np.ndarray[floating, ndim=1, mode="c"] X_data,
 
     for i in xrange(n_features):
         variances[i] += (n_samples - counts_nan[i] - counts[i]) * means[i]**2
-        variances[i] /= (n_samples - counts_nan[i])
+        # int(n_samples) is necessary to convert from unsigned int to any int
+        if int(n_samples) - ddof - counts_nan[i] > 0:
```

Comment:
Just out of curiosity do you really get a compilation error without `int()`? I thought Cython would know to how to cast in simple cases like this.

Reply:
I don't think Cython has any issue with casting since all three (`n_samples`, `ddof` and `counts_nan`) are all `unsigned longs`. The issue is more semantic: since all of these are `unsigned`, the result will underflow in the event that the condition is actually <0, resulting in an `INT_MAX`-like number, which is >>0. Therefore it is necessary to explicitly cast to an int, so that we can actually get negative numbers, so the check properly works.